### PR TITLE
Added Teletaster Protocol

### PR DESCRIPTION
**Pull request by ramyres**
Adds support for Tedsen Teletaster protocol.

### CODE REVIEW

File: libs/pilight/protocols/433.92/teletaster.c [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	teletaster->message = json_mkobject();
+	json_append_member(teletaster->message, "id", json_mkstring(id));}
+
+static void parseCode(void) {
+	int x = 0, i = 0;
+	int len = (int)((double)AVG_PULSE_LENGTH*((double)PULSE_MULTIPLIER/2));
+	char id[ID_LENGTH] = "abcdefghi";
+
+
+
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/teletaster.c]: Can you remove these bogus lines?

File: libs/pilight/protocols/433.92/teletaster.c [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+static void createFooter(void) {
+	teletaster->raw[35]=(PULSE_DIV*AVG_PULSE_LENGTH);
+}
+
+static int createCode(struct JsonNode *code) {
+	char *id = NULL;
+	int i = 0;
+	//double itmp = -1;
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/teletaster.c]: Can you remove this?

File: libs/pilight/protocols/433.92/teletaster.c [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+		teletaster->rawlen = RAW_LENGTH;
+	}
+	return EXIT_SUCCESS;
+}
+
+static int validateId(char *id) {
+	int i = 0;
+
+	for(i=0; i<ID_LENGTH; i++) {
+		if(*(id+i) != '1' && *(id+i) != '0' && *(id+i) != 'x' && *(id+i) != 'X')
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/teletaster.c]: Can you add curly brackets here?

File: libs/pilight/protocols/433.92/teletaster.c [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+		createMessage(id);
+		clearCode();
+		createId(id);
+		createFooter();
+		teletaster->rawlen = RAW_LENGTH;
+	}
+	return EXIT_SUCCESS;
+}
+
+static int validateId(char *id) {
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/teletaster.c]: This function isn't used anywhere.

File: libs/pilight/protocols/433.92/teletaster.c [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+			return 0;
+		}
+	}
+
+	return -1;
+}
+
+static void createMessage(char* id) {
+	teletaster->message = json_mkobject();
+	json_append_member(teletaster->message, "id", json_mkstring(id));}
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/teletaster.c]: Where are the on/off messages?

File: libs/pilight/protocols/433.92/teletaster.c [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+static void parseCode(void) {
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/teletaster.c]: Where is the on/off decoding?

File: libs/pilight/protocols/433.92/teletaster.c [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+	for(i=0; i<ID_LENGTH; i++) {
+		if(*(id+i) != '1' && *(id+i) != '0' && *(id+i) != 'x' && *(id+i) != 'X')
+			return 0;
+	}
+	return 1;
+}
+
+
+
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/teletaster.c]: Bogus lines

**Comment by CurlyMoo** [comment]
1. Please give it a MPLv2 license header.
2. Please include a documentation page.

File: libs/pilight/protocols/433.92/teletaster.c (hunk 1 above)
Comment (ramyres) [reply libs/pilight/protocols/433.92/teletaster.c]: ok

File: libs/pilight/protocols/433.92/teletaster.c (hunk 2 above)
Comment (ramyres) [reply libs/pilight/protocols/433.92/teletaster.c]: ok

File: libs/pilight/protocols/433.92/teletaster.c (hunk 3 above)
Comment (ramyres) [reply libs/pilight/protocols/433.92/teletaster.c]: sure

File: libs/pilight/protocols/433.92/teletaster.c (hunk 4 above)
Comment (ramyres) [reply libs/pilight/protocols/433.92/teletaster.c]: ok, deleted

File: libs/pilight/protocols/433.92/teletaster.c (hunk 6 above)
Comment (ramyres) [reply libs/pilight/protocols/433.92/teletaster.c]: There ise non. May I refer you again to https://forum.pilight.org/Thread-Teletaster-protocol

File: libs/pilight/protocols/433.92/teletaster.c (hunk 7 above)
Comment (ramyres) [reply libs/pilight/protocols/433.92/teletaster.c]: removed

**Comment by ramyres** [comment]
On the license header and the doc page, can you point me to an example of how this should look like? I tried to spot one in the code, but didn't find one. I derived the "template" for the protocol from your arctech_switch.

**Comment by CurlyMoo** [comment]
https://github.com/pilight/pilight/pull/389
https://github.com/pilight/pilight/blob/rewrite/libs/pilight/protocols/433.92/arctech_switch.c

File: libs/pilight/protocols/433.92/teletaster.c (hunk 6 above)
Comment (CurlyMoo) [reply libs/pilight/protocols/433.92/teletaster.c]: Does a `contact` not fit better?

Comment (ramyres) [reply libs/pilight/protocols/433.92/teletaster.c]: When I implemented this last year, I was scanning through the existing protocols and had the impression that `contact` is a "passive" element that only reports its state, but cannot be actively triggered to change it. This Teletaster is like a push button: You send the signal and toggle the state, but you can't derive the current state of the switch.

**Pull request closed without merging**